Source/WebCore/html/FileInputType.cpp

```diff
@@ -276,7 +224,11 @@ void FileInputType::createShadowSubtree()
     ASSERT(element());
     ASSERT(element()->shadowRoot());
 
-    auto button = element()->multiple() ? UploadButtonElement::createForMultiple(element()->document()) : UploadButtonElement::create(element()->document());
+    auto button = HTMLInputElement::create(inputTag, element()->document(), nullptr, false);
+    ScriptDisallowedScope::EventAllowedScope eventAllowedScopeBeforeAppend { button };
```

Comment:
To limit the scope you could either wrap this code inside brackets:
```
    auto button = HTMLInputElement::create(inputTag, element()->document(), nullptr, false);
    {
        ScriptDisallowedScope::EventAllowedScope eventAllowedScopeBeforeAppend { button };
        button->setType(InputTypeNames::button());
        button->setPseudo(ShadowPseudoIds::fileSelectorButton());
        button->setValue(element()->multiple() ? fileButtonChooseMultipleFilesLabel() : fileButtonChooseFileLabel());
    }
 ```

or inside a lambda. Otherwise the scope exception will be extended until the end of the method.